ui/components-react/src/components-react/editors/EditorContainer.tsx

```diff
@@ -130,78 +145,104 @@ export class EditorContainer extends React.PureComponent<EditorContainerProps> {
     }
 
     return clonedNode;
-  }
+  };
 
-  private _handleEditorBlur = (_e: React.FocusEvent) => {
+  /** Event Handlers
+   * @internal
+   */
+  const handleCancel = () => onCancel();
+  const handleClick = (e: React.MouseEvent) => e.stopPropagation();
+  const handleContainerBlur = (e: React.FocusEvent) => e.stopPropagation();
+  const handleEditorCommit = (args: PropertyUpdatedArgs): void =>
+    void commit(args);
+
+  const handleContainerCommit = async (): Promise<void> => {
+    const newValue = editorRef && (await editorRef.current?.getPropertyValue());
     // istanbul ignore else
-    if (
-      !this.props.ignoreEditorBlur &&
-      this._propertyEditor &&
-      this._propertyEditor.containerHandlesBlur
-    )
-      void this._handleContainerCommit();
+    if (newValue !== undefined) {
+      void commit({
+        propertyRecord,
+        newValue,
+      });
+    }
   };
 
-  private _handleContainerBlur = (e: React.FocusEvent) => {
-    e.stopPropagation();
+  const onPressEscape = (): void => {
+    // istanbul ignore else
+    if (propertyEditor && propertyEditor.containerHandlesEscape) handleCancel();
+  };
+
+  const onPressEnter = (e: React.KeyboardEvent): void => {
+    // istanbul ignore else
+    if (propertyEditor && propertyEditor.containerHandlesEnter) {
```

Comment:
Maybe we can write `propertyEditor?. propertyEditor.containerHandlesEnter` for a cleaner solution? You should make this change to all similar places

Reply:
yeah thats a good catch. i'll change it in the conditionals that use that syntax